menu changes

Kelvin285 opened the pull request:
I think I've fixed my github fork!  Had to delete everything and fork it again.  Here's the menu changes with the custom title screen background.

Kelvin285 committed:
menu changes

I think I've fixed my github fork!  Had to delete everything and fork it again.  Here's the menu changes with the custom title screen background.

bconlon1 committed:
Test

bconlon1 committed:
refactor: Cleanup unused imports.

bconlon1 commented on src/main/java/com/gildedgames/aether/client/event/listeners/OverlayListener.java:
```diff
@@ -0,0 +1,68 @@
+        AetherWorldDisplayHelper.loadedLevel = Minecraft.getInstance().level;
+
+        var player = minecraft.getSingleplayerServer().getPlayerList().getPlayers().get(0);
+        var world = (ServerLevel)player.getCommandSenderWorld();
```
This variable seems to be unused, not sure if there's any reason in leaving it in.

bconlon1 commented on src/main/java/com/gildedgames/aether/client/gui/screen/menu/AetherTitleScreen.java:
```diff
@@ -65,22 +66,28 @@ protected void init() {
 			}
 			float f = this.fading ? (float) (Util.getMillis() - this.fadeInStart) / 1000.0F : 1.0F;
-			this.panorama.render(partialTicks, Mth.clamp(f, 0.0F, 1.0F));
+			if (AetherWorldDisplayHelper.loadedLevel == null || !AetherConfig.CLIENT.enable_world_preview.get() || Minecraft.getInstance().level == null) this.panorama.render(partialTicks, Mth.clamp(f, 0.0F, 1.0F));
```
Might be better to just wrap the entire panorama render code in this check instead of the blits.

bconlon1 replied:
Actually I'm wondering if this is even necessary if you have the mixin.

Kelvin285 replied:
yeah I added the mixin later and forgot to remove that part of the code.  I'll remove it now.  It doesn't really need to stay there anymore.

bconlon1 commented on src/main/java/com/gildedgames/aether/client/gui/screen/menu/AetherWorldDisplayHelper.java:
```diff
@@ -0,0 +1,154 @@
+    }
+
+    public static void fixWorld() {
+        Minecraft minecraft = Minecraft.getInstance();
```
Unused?

bconlon1 commented on src/main/java/com/gildedgames/aether/client/gui/screen/menu/AetherWorldDisplayHelper.java:
```diff
@@ -0,0 +1,154 @@
+            Minecraft minecraft = Minecraft.getInstance();
+            minecraft.getSoundManager().play(SimpleSoundInstance.forUI(SoundEvents.UI_BUTTON_CLICK, 1.0F));
+            if (minecraft.getLevelSource().levelExists(loadedSummary.getLevelId())) {
+                //this.minecraft.forceSetScreen(new GenericDirtMessageScreen(new TranslatableComponent("selectWorld.data_read")));
```
If this is no longer needed it can probably be removed.

Kelvin285 replied:
didn't even notice that part of the code was there.  I copied those three lines over from another class and modified them.  I was wondering why that sound kept playing lol

bconlon1 replied:
I meant to just point out the commented out code. Thank you GitHub.

bconlon1 commented on src/main/java/com/gildedgames/aether/core/mixin/mixins/client/MinecraftMixin.java:
```diff
@@ -30,10 +33,50 @@ public void getSituationalMusic_Dimension(CallbackInfoReturnable<Music> cir) {
+    @Inject(at=@At(value = "HEAD"), method = "clearLevel(Lnet/minecraft/client/gui/screens/Screen;)V")
+    public void clearLevel(Screen pScreen, CallbackInfo info) {
+        if (!AetherWorldDisplayHelper.loadingLevel) {
+            AetherWorldDisplayHelper.loadingLevel = false;
```
This seems redundant if loadingLevel is already false, unless I'm misunderstanding something here.

Kelvin285 replied:
The menu variables should only be set to null when leaving the menu screen, but clearLevel is called whenever a world is loaded so I have to put that there otherwise it will clear all the menu variables while it's loading a world file for the menu background and then you won't be able to see anything.

bconlon1 replied:
I was meaning to only point out the loadingLevel = false line.

bconlon1 commented on src/main/java/com/gildedgames/aether/core/mixin/mixins/client/MinecraftMixin.java:
```diff
@@ -30,10 +33,50 @@ public void getSituationalMusic_Dimension(CallbackInfoReturnable<Music> cir) {
+                AetherWorldDisplayHelper.fixWorld();
+                Minecraft minecraft =  Minecraft.getInstance();
+                minecraft.getSingleplayerServer().halt(false);
+                //minecraft.getSingleplayerServer().stopServer();
```
Is this code needed anymore?

Kelvin285 replied:
if I don't halt the integrated server when switching between levels the game crashes because the menu world has to be loaded without a session lock.  Halting the integrated server gives the game time to clear everything before switching to the next world and restarting the server.

bconlon1 replied:
I was meaning to only point out the commented code.

bconlon1 reviewed (changes requested):
In general you also have various NPE warnings for calling certain methods that should maybe have checks for safety; even if vanilla works fine with them there's no telling what other mods will do.

bconlon1 commented:
There's some minor formatting critiques I have as well but I can clean those up myself later since they're nitpicky.

Kelvin285 committed:
More menu changes

- chat is no longer visible when you're viewing a level through the title screen
- fixed a bug where disabling and re-enabling title screen world view would set you into spectator mode
- fixed a bug where you couldn't delete or make new worlds while viewing a world in the title screen

Kelvin285 committed:
Merge branch 'feat/bconlon/menu' of https://github.com/Kelvin285/The-Aether into feat/bconlon/menu

Kelvin285 committed:
menu changes (part 3)

fixed world saving/loading (probably)

bconlon1 committed:
feat: Cleanup to the menu code.

bconlon1 reviewed (approved):
lgtm
